compare_attempts.py

```diff
@@ -0,0 +1,120 @@
+import csv
+import os
+import statistics
+import sys
+from typing import List
+
+import pandas as pd
+from dotenv import load_dotenv
+
+import utils
+from project_types.custom_types import Approach, Task, bcolors
+
+
+def compare(data_folder_path: str, samples_per_task: int):
+    header = [
+        "Filename",
+        "Total Tasks",
+        "Total Samples",
+        "Vulnerable Samples",
+        "Expected CWE Samples",
+        "Sample Vulnerable Percentages",
+        "Min Vulnerable Percentage",
+        "Median Vulnerable Percentage",
+        "Average Vulnerable Percentage",
+        "Max Vulnerable Percentage",
+        "Min Expected CWE Percentage",
+        "Median Expected CWE Percentage",
+        "Average Expected CWE Percentage",
+        "Max Expected CWE Percentage",
+    ]
+
+    matrix = []
+
+    for file in os.listdir(data_folder_path):
+        data_file_path = os.path.join(data_folder_path, file)
+        # checking if it is a file
+        if os.path.isfile(data_file_path):
+            print(f"Analyzing: {data_file_path}")
+            approach = utils.read_approaches_file(data_file_path)
+            if (
+                not approach.vulnerable_percentage is None
+                and not approach.expected_cwe_percentage is None
+            ):
+                results = []
+                matrix.append(results)
+                analyze(approach, samples_per_task, results)
+            else:
+                print(
+                    f"{bcolors.FAIL}approach is not analyzed yet, analyze it first{bcolors.ENDC}",
+                    file=sys.stderr,
+                )
+
+    # sort by column 2, "Vulnerable Samples"
+    matrix.sort(key=lambda row: row[2])
```

Comment:
row[2] is Total Samples, not Vulnerable Samples
to avoid this, use row["Vulnerable Samples"] instead (this requires making results a dict instead of an array)

Reply:
fixed using dict